### ISSUE TITLE
Improvements

### ISSUE THREAD

**Issue by snowypowers** [CityOfZion/neon-js#16]
## Lint

I see a PR for linting but it does not introduce any code standards. The current lint is not working due to absence of a .eslintrc file.

I would like to use [StandardJS](https://standardjs.com/) as the default style. It is clean and proven.

## Refactor+Cleanup

Do a general cleanup and refactor. There is a lot of rough code and unused deps that should be cleaned.  A refactoring will allow us to restructure the code before we start writing more testcases.

## Testing

We need more tests.

## JS vs TS

While it is nice to have both repos, I think that it is a duplicated effort to maintain both repos considering that they offer the same thing in the end (a JS file). I think we should maintain only one for the sake of consistency.

Any thoughts?

**Comment by ixje** [comment]
I like the suggestion of using a coding standard. I had a look at the StandardJS mentioned and it looks fine to me. I personally prefer 4 spaces over 2 because otherwise it looks so condensed, but that's about it what I don't like. I do feel there could be some additions like the following: 

- no magic numbers. 

Instead of
```
export const getAccountsFromWIFKey = ($WIFKey) => {
	var privateKey = getPrivateKeyFromWIF($WIFKey);
	if (privateKey == -1 || privateKey == -2) {
		return privateKey;
	}

	return getAccountsFromPrivateKey(privateKey);
};
```
do
```
export const getAccountsFromWIFKey = ($WIFKey) => {
	var privateKey = getPrivateKeyFromWIF($WIFKey);
    const ENCODING_ERROR = -1
    const WIF_VERIFY_FAIL = -2
	if (privateKey == ENCODING_ERROR || privateKey == WIF_VERIFY_FAIL) {
		return privateKey;
	}

	return getAccountsFromPrivateKey(privateKey);
};
```

- use single return statements
- have guidelines how to name functions/variables. E.g. use get/set instead of retrieve/fetch/collect. Like now there's a function called `getAccountsFromPublicKey ` and `fetchAccountsFromPublicKeyEncoded`. The latter should be `fetchAccountsFromPublicKeyEncoded` imo.  
- prefix functions that return a boolean with `is` e.g. `isValidInput()` vs `InputValid()` 

Exact rules for the above should be looked up, as I know they're in some coding standard with explanations why they make sense. I'm just writing this quickly from the top of my head.

**Comment by snowypowers** [comment]
When I selected StandardJS, it was due to its simplicity to drop into any project. If I wanted to adjust any of its rules, I might look at writing my own config file instead.

- Yes, I have seen the magic numbers and do agree that something has to be done. However, I rather prefer throwing errors than returning negative numbers. The catching can be done at the higher level (api.js). i havn't done anything about that because it might impact Neon Wallet if we start throwing errors instead of numbers.

- I don't see any fetches?

- There is only 2 functions that returns bool and they start with `verify`. It is consistent for now but a rename might be valid.

**Comment by ixje** [comment]
Don't get me wrong, I'm would be totally fine with using StandardJS. It's like 99% how I already work :)

- I agree we should throw errors. I mentioned it in #21 as well. I'm sure it will break neon-wallet as I've been digging through that as a reference when creating a mobile wallet.
- It could very well be that there are no fetches in the current code. I should not have mixed examples behind the proposed rule with actual examples found in code as this leads to confusion.
- It might be ok now, which is great. The intention was to keep it ok by having a guideline incase more functions get added :)
